### PR TITLE
Trims the buildbox

### PR DESCRIPTION
**Pull request by tcsc**
Enables the docker BuildKit in an attempt to speed up builds, and trims just under 2GB off the image size.

Prior to these changes, `docker image ls` reported the image as 6.08GB after a local build, and ~~5.06GB~~ 4.17GB after

See-Also: #6149

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -107,10 +110,12 @@ RUN apt-get update -y --fix-missing && \
     rm -rf /var/lib/apt/lists/*
 
 # Install gcloud SDK and Firestore emulator.
-RUN curl -sSL https://sdk.cloud.google.com | bash -s -- --install-dir=/opt --disable-prompts
+RUN (curl -sSL https://sdk.cloud.google.com | bash -s -- --install-dir=/opt --disable-prompts) && \
+    rm -rf /opt/google-cloud-sdk/.install/.backup
 ENV PATH="$PATH:/opt/google-cloud-sdk/bin"
 # beta component is required by firestore emulator: https://cloud.google.com/sdk/gcloud/reference/beta/emulators/firestore/start
-RUN gcloud components install cloud-firestore-emulator beta
+RUN gcloud components install cloud-firestore-emulator beta && \
```
Comment (webvictim) [inline build.assets/Dockerfile:117]: Maybe set the PATH up front and condense these two steps into one, so we only remove the cache once?

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -172,11 +178,11 @@ RUN (export PROTOC_TARBALL=protoc-${PROTOC_VER}-linux-$(if [ "$BUILDARCH" = "amd
      cd /tmp && unzip /tmp/${PROTOC_TARBALL} -d /usr/local && \
      chmod -R a+r /usr/local/include/google/protobuf && \
      chmod -R a+xr /usr/local/bin/protoc && \
-     rm /tmp/${PROTOC_TARBALL})
-RUN (git clone https://github.com/gogo/protobuf.git ${GOPATH}/src/github.com/gogo/protobuf && \
+     rm /tmp/${PROTOC_TARBALL})
```
Comment (webvictim) [inline build.assets/Dockerfile:181]: Random whitespace added here

**Review by zmb3 — APPROVED** [review]
LGTM with Gus's suggestions addressed.

**Review by r0mant — COMMENTED** [review]
Yep, we need to do this for Centos 7 dockerfile and makefile command also. We're going to use Centos 7 for all amd64 artifacts for Teleport 10.

File: build.assets/Dockerfile (hunk 2 above)
Comment (tcsc) [reply build.assets/Dockerfile]: Removed

File: build.assets/Dockerfile (hunk 1 above)
Comment (tcsc) [reply build.assets/Dockerfile]: Done.

**Review by nklaassen — APPROVED** [review]

File: build.assets/Makefile [hunk 3]
```diff
@@ -121,7 +121,7 @@ build-binaries-fips: buildbox-fips
 buildbox:
 	if [[ "$(BUILDBOX_NAME)" == "$(BUILDBOX)" ]]; then \
 		if [[ $${DRONE} == "true" ]] && ! docker inspect --type=image $(BUILDBOX) 2>&1 >/dev/null; then docker pull $(BUILDBOX) || true; fi; \
-		docker build --platform=linux/$(RUNTIME_ARCH) \
+		 DOCKER_BUILDKIT=1 docker build --platform=linux/$(RUNTIME_ARCH) \
```
Comment (webvictim) [inline build.assets/Makefile:124]: There’s also a leading space here; doesn’t make any difference to the way it runs, just can’t unsee it 😁

**Review by webvictim — APPROVED** [review]

**Pull request merged**